src/main/scala/com/mozilla/telemetry/streaming/EventPingEvents.scala

```diff
@@ -0,0 +1,149 @@
+      .option("spark.streaming.kafka.consumer.cache.maxCapacity", kafkaCacheMaxCapacity)
+      .option("subscribe", TelemetryKafkaTopic)
+      .option("startingOffsets", opts.startingOffsets())
+      .load()
+
+    val outputPath = opts.outputPath()
+
+    explodeEvents(pings.select("value"))
+      .repartition(1)
+      .writeStream
+      .queryName(QueryName)
+      .format("parquet")
+      .option("path", s"${outputPath}/${outputPrefix}")
+      .option("checkpointLocation", opts.checkpointPath())
+      .partitionBy("submission_date_s3", "docType")
+      .start()
+      .awaitTermination()
+
+  }
+
+  def writeBatchEvents(spark: SparkSession, opts: EventPingEvents.Opts): Unit = {
+    implicit val sc = spark.sparkContext
+    import spark.implicits._
+
+    datesBetween(opts.from(), opts.to.get).foreach { currentDate =>
+      val pings = MozDataset("telemetry")
+        .where("sourceName") { case "telemetry" => true }
+        .where("docType") { case docType if allowedDocTypes.contains(docType) => true }
+        .where("appName") { case appName if allowedAppNames.contains(appName) => true }
+        .where("submissionDate") { case date if date == currentDate => true }
+        .records(opts.fileLimit.get)
+        .map(_.toByteArray)
+
+      val pingsDataframe = pings.toDF("value")
+
+      val outputPath = s"${opts.outputPath()}/${outputPrefix}/submission_date_s3=$currentDate/docType=event"
+
+      logger.info(s"Processing and event pings for $currentDate")
+
+      explodeEvents(pingsDataframe)
+        .write
+        .option("maxRecordsPerFile", opts.maxRecordsPerFile())
+        .mode("overwrite")
+        .parquet(outputPath)
+
+      logger.info(s"Event ping events written for $currentDate")
+    }
+
+    spark.stop()
+  }
+
+  private[streaming] def explodeEvents(messages: DataFrame): Dataset[EventRow] = {
+    import messages.sparkSession.implicits._
+
+    messages.flatMap(v => {
+      //try {
+        val m = Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]])
+        val fields = m.fieldsAsMap
+        val docType = fields.getOrElse("docType", "").asInstanceOf[String]
+        if (!allowedDocTypes.contains(docType)) {
+          None
+        } else {
+          val ping = EventPing(m)
+          ping.processEventMap.flatMap { case (process, events) =>
+            events.map { e =>
+              EventRow(ping.meta.documentId.get, ping.meta.clientId.get, ping.meta.normalizedChannel, ping.meta.geoCountry,
+                ping.getLocale, ping.meta.appName, ping.meta.appVersion, ping.getOsName, ping.getOsVersion,
+                ping.payload.sessionId, ping.payload.subsessionId, ping.sessionStart,
+                (ping.meta.Timestamp / 1e9).toLong, ping.meta.sampleId.map(_.toString), ping.getMSStyleExperiments,
+                e.timestamp, e.category, e.method, e.`object`, e.value, e.extra, process)
+
+            }
+          }
+        }
+      /*
+      } catch {
+        // TODO: track parse errors
```

Comment:
What's current behavior if we have parse errors? Does the whole job shut down?

Reply:
That's some copy-pasta from using the streaming experiment enrollments job as a base :) Because this job can run as a streaming job as well as in batch mode, we don't want to throw on parse errors and kill the job but also we probably start tracking errors and logging them instead of eating them silently